pulp_container/app/registry_api.py

```diff
@@ -300,6 +300,15 @@ def get_dr_push(self, request, path, create=False):
                 repository = repository.cast()
                 if not repository.PUSH_ENABLED:
                     raise RepositoryInvalid(name=path, message="Repository is read-only.")
+            elif create:
+                with transaction.atomic():
+                    repo_serializer = serializers.ContainerPushRepositorySerializer(
+                        data={"name": path}, context={"request": request}
+                    )
+                    repo_serializer.is_valid(raise_exception=True)
+                    repository = repo_serializer.create(repo_serializer.validated_data)
+                    distribution.repository = repository
+                    distribution.safe()
```

Comment:
From IRC:
<x9c4> So you would create a private empty distro and then when pushing to it you decide it's fate to serve a push repo? Sounds fine to me.
<ipanova> yes, that would be my take on it